Content.Client/HealthAnalyzer/UI/HealthAnalyzerWindow.xaml.cs

```diff
@@ -1,76 +1,203 @@
+using System.Linq;
 using System.Text;
 using Content.Shared.Damage;
 using Content.Shared.Damage.Prototypes;
 using Content.Shared.FixedPoint;
 using Content.Shared.IdentityManagement;
 using Content.Shared.MedicalScanner;
+using Content.Shared.Mobs.Components;
 using Robust.Client.AutoGenerated;
 using Robust.Client.UserInterface.CustomControls;
 using Robust.Client.UserInterface.XAML;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.UserInterface.Controls;
 using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
 
 namespace Content.Client.HealthAnalyzer.UI
 {
     [GenerateTypedNameReferences]
     public sealed partial class HealthAnalyzerWindow : DefaultWindow
     {
+        private readonly IEntityManager _entityManager;
+        private readonly SpriteSystem _spriteSystem;
+        private readonly IPrototypeManager _prototypes;
+
+        private bool _isSettledWidth = false;
+
+        private const int AnalyzerHeight = 430;
+        private const int AnalyzerMaxWidth = 500;
+
         public HealthAnalyzerWindow()
         {
             RobustXamlLoader.Load(this);
+
+            _entityManager = IoCManager.Resolve<IEntityManager>();
+            _spriteSystem = _entityManager.System<SpriteSystem>();
+            _prototypes = IoCManager.Resolve<IPrototypeManager>();
+
+            RootContainer.OnResized += OnRootContainerResized;
         }
 
         public void Populate(HealthAnalyzerScannedUserMessage msg)
         {
-            var text = new StringBuilder();
-            var entities = IoCManager.Resolve<IEntityManager>();
+            NoPatientDataText.Visible = true;
+            GroupsContainer.RemoveAllChildren();
 
-            if (msg.TargetEntity != null && entities.TryGetComponent<DamageableComponent>(msg.TargetEntity, out var damageable))
+            if (msg.TargetEntity == null
+                || !_entityManager.TryGetComponent<DamageableComponent>(msg.TargetEntity, out var damageable))
             {
-                string entityName = "Unknown";
-                if (msg.TargetEntity != null &&
-                    entities.TryGetComponent<MetaDataComponent>(msg.TargetEntity.Value, out var metaData))
-                    entityName = Identity.Name(msg.TargetEntity.Value, entities);
+                return;
+            }
+
+            NoPatientDataText.Visible = false;
+
+            patientDamageAmount.Text = Loc.GetString(
+                "health-analyzer-window-entity-damage-total-text",
+                ("amount", damageable.TotalDamage)
+            );
+
+            var entityName = "Unknown";
```

Comment:
Not localised.

Reply:
![image](https://github.com/space-wizards/space-station-14/assets/2144635/7f6ff848-b738-4610-ab14-6619c91dc21c)